### PR TITLE
Test with Chef 12, upgrade chefDK on Travis, fix local Ubuntu 16.04 testing

### PR DESCRIPTION
**Pull request by aried3r**
### Description

Tests the cookbook with Chef 12 as well. Mainly so we can verify this cookbook keeps working once #161 is fixed.

What I'm wondering is how this currently passes on master, I see [the test run](https://travis-ci.org/sous-chefs/ruby_rbenv/jobs/243763740) but also the `Test Summary: 0 successful, 0 failures, 0 skipped`, so I'm not sure if it's doing anything at all. Since I'm new to kitchen, maybe somebody can resolve this for me?

### Issues Resolved

_none_

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by aried3r** [comment]
~~Heh, I'm actually facing a different error now. Nothing is tested with Chef 13, see the latest successful test run, which should pick the latest chef https://travis-ci.org/sous-chefs/ruby_rbenv/jobs/244500052#L333 but pics 12.19.~~

~~Could it be because of https://github.com/sous-chefs/ruby_rbenv/blob/master/.travis.yml#L7 or is it something else entirely?~~

File: providers/ruby.rb [hunk 1]
```diff
@@ -81,8 +81,6 @@ def perform_install
 
     Chef::Log.debug("#{new_resource} build time was " \
       "#{(Time.now - install_start) / 60.0} minutes")
-
-    new_resource.updated_by_last_action(true)
```
Comment (aried3r) [inline providers/ruby.rb:85]: I removed one occurrence of [FC085](http://www.foodcritic.io/#FC085) but there is one more in [this file](https://github.com/sous-chefs/ruby_rbenv/blob/56b6ed3af0362d72ab1f6593ace45dcece74630e/libraries/chef_rbenv_script_helpers.rb#L59-L62) which gets called [a few times](https://github.com/sous-chefs/ruby_rbenv/search?utf8=✓&q=set_updated&type=) but I'm not sure if I can just remove it or if it's been replaced by something different. I saw [other cookbooks](https://github.com/chef-cookbooks/aws/commit/7c3ed4f5178e736e2db245155130928b66047314) just [removed it](https://github.com/dhoer/chef-nssm/commit/5de5e61c877abec9acab490e60baf5658ec1e807) but I haven't found any cookbook using it with something like `set_updated` in this one.

File: .kitchen.yml [hunk 2]
```diff
@@ -5,16 +5,27 @@ provisioner:
...
+      box: bento/ubuntu-16.04
+    provisioner:
+      require_chef_omnibus: 12
+    transport:
+      username: ubuntu
 
 suites:
 - name: system_ruby
   run_list:
+  - recipe[apt]
```
Comment (aried3r) [inline .kitchen.yml:28]: Without this the local image wouldn't find some packages and fail.

File: .kitchen.yml [hunk 3]
```diff
@@ -5,16 +5,27 @@ provisioner:
...
   - name: ubuntu-14.04
   - name: ubuntu-16.04
-  - name: fedora-25
+  - name: ubuntu-16.04-chef-12
+    driver_config:
+      box: bento/ubuntu-16.04
+    provisioner:
+      require_chef_omnibus: 12
+    transport:
+      username: ubuntu
```
Comment (aried3r) [inline .kitchen.yml:23]: The `vagrant` user is not the correct one for the ubuntu-16.04 image.

File: providers/ruby.rb (hunk 1 above)
Comment (tas50) [reply providers/ruby.rb:85]: You could do it now or we could clean it up later, but the this whole thing should get added to a converge_by block so we get nice messaging. Right now we're using the Log class do to this which isn't quite as nice.

**Comment by tas50** [comment]
Solid start to getting this in place. Thanks

**Pull request merged**